### ISSUE TITLE
SwiftLintPlugin is not working with the included path from yml file

### ISSUE THREAD

**Issue by miku1958** [realm/SwiftLint#4877]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I add included path in yml file but SwiftLintPlugin doesn't work with it:

### Environment

swiftlint_version: 0.51.0

```yml
included:
    - ./../Module
```

**Comment by miku1958** [comment]
I remove this line and it works
https://github.com/realm/SwiftLint/blob/bb11e6a0a7fc4baeeefe95485fdc1b849362d331/Plugins/SwiftLintPlugin/SwiftLintPlugin.swift#L44

**Comment by SimplyDanny** [comment]
Passing paths explicitly to the command ignores the configuration for included directories/files.

**Comment by miku1958** [comment]
Thanks for the reply @SimplyDanny, I think we should remove this:
https://github.com/realm/SwiftLint/blob/bb11e6a0a7fc4baeeefe95485fdc1b849362d331/Plugins/SwiftLintPlugin/SwiftLintPlugin.swift#L44

**Comment by SimplyDanny** [comment]
A build plugin runs per target and in the case of SwiftLint only lints files that are part of the target. If we remove this line, the plugin would always lint everything which makes the operation quite expensive when only a specific target is actually built.

**Comment by miku1958** [comment]
I see, is there any solution?

**Comment by SimplyDanny** [comment]
Let's think about what these plugins are supposed to do:

They run for a specific target of your project - a subset of your code. They run in their own sandbox with a copy of the files in the target they are allowed to see. They are not allowed to change any of the files in your code base.

So all in all, SPM/Xcode creates a very restrictive environment for the plugin to run in. If we now consider to lint files that are outside of the current target and even located in a parent folder (`../Module`) I don't think this would be in the sense of the SPM/Xcode build mechanism.

That's why SwiftLint takes `excluded` files/folders specified in the configuration into account, but not additional `included` locations.

However, if the files located inside of `../Module` are part of one of the targets in your project, they will be linted at some point in case the plugin is run for all of them.

**Comment by miku1958** [comment]
We split the code into different targets. When we run the main target, all the targets are built, but only the files in the main target are covered.

**Comment by miku1958** [comment]
I just found the reason, we only add the sub targets as libs but not actually "build" them in scheme, added them into scheme's build tab and fixed this issue
